app/models/topic.rb

```diff
@@ -910,6 +910,7 @@ def move_posts(moved_by, post_ids, opts)
     post_mover = PostMover.new(self, moved_by, post_ids)
 
     if opts[:destination_topic_id]
+      DiscourseEvent.trigger(:topic_merged,  post_mover.original_topic, Topic.find_by(id: opts[:destination_topic_id]))
```

Comment:
> Topic.find_by(id: opts[:destination_topic_id])

This adds an extra database query. You can trigger this event only after `post_mover.to_topic opts[:destination_topic_id]` and then call `post_mover.destination_topic`